src/App.js

```diff
@@ -10,28 +10,161 @@ import Results from "./UIComponents/Results";
 import Predict from "./UIComponents/Predict";
 import SaveModel from "./UIComponents/SaveModel";
 import { styles } from "./constants";
+import { connect } from "react-redux";
+import {
+  setCurrentPanel,
+  getPanelVisible,
+  getPanelEnabled,
+  panelList,
+  validationMessages
+} from "./redux";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { faSquare, faCheckSquare } from "@fortawesome/free-regular-svg-icons";
 
-export default class App extends Component {
+class PanelTabs extends Component {
   static propTypes = {
-    mode: PropTypes.object,
+    currentPanel: PropTypes.string,
+    setCurrentPanel: PropTypes.func
+  };
+
+  getTabStyle(panel) {
+    if (getPanelEnabled(panel)) {
+      if (panel === this.props.currentPanel) {
+        return { ...styles.tab, ...styles.currentTab };
+      } else {
+        return styles.tab;
+      }
+    } else {
+      return { ...styles.tab, ...styles.disabledTab };
+    }
+  }
+
+  render() {
+    const { setCurrentPanel } = this.props;
+
+    return (
+      <div style={styles.tabContainer}>
+        {panelList
+          .filter(panel => {
+            return getPanelVisible(panel.id);
+          })
+          .map(panel => {
+            return (
+              <div
+                key={panel.id}
+                style={this.getTabStyle(panel.id)}
+                onClick={() => setCurrentPanel(panel.id)}
+              >
+                {panel.label}
+              </div>
+            );
+          })}
+      </div>
+    );
+  }
+}
```

Comment:
Right now we have 1 component per file; is there an advantage to putting multiple in this parent file?

Reply:
Just a slight personal preference for not creating new files in the case where they remain small, and have no external consumers.